src/resolve.rs

```diff
@@ -75,26 +75,17 @@ fn system_nameservers() -> io::Result<Option<Nameserver>> {
         }
     };
 
-    let first_adapter = match adapters.first() {
-        Some(a) => a,
-        None => {
-            warn!("No network adapters available");
-            return Ok(None);
-        }
-    };
-    debug!("Found network adapter {:?}", first_adapter.adapter_name());
-
-    let first_nameserver = match first_adapter.dns_servers().first() {
-        Some(ns) => ns,
-        None => {
-            warn!("No nameservers available");
-            return Ok(None);
-        }
-    };
-    debug!("Found nameserver {:?}", first_nameserver);
+    for dns_server in adapters
+        .iter()
+        .flat_map(|adapter| adapter.dns_servers().iter()) {
+            if dns_server.is_ipv4() {
```

Comment:
What if current DNS server has external IPv6 address? Maybe better decision will be check response from DNS server on error.

Reply:
> What if current DNS server has external IPv6 address? Maybe better decision will be check response from DNS server on error.

I follow the same logic on unix https://github.com/ogham/dog/blob/master/src/resolve.rs#L53

The IPv6 issue can be dealt with in another pull request.

Reply:
The idea behind the code in this file is to get the correct IP address of the DNS server, not check if it is IPv4 or IPv6. This could potentially lead to the same error in the future.